docroot/themes/humsci/su_humsci_theme/js/hover_menu.js

```diff
@@ -29,25 +29,51 @@
         }
       }
 
+      // Check main menu items to ensure dropdown submenus don't get orphaned or cut off.
+      function menuEdgeCheck() {
+        var $viewportWidth = $('body').innerWidth();
+        $('ul.decanter-nav-primary > li', context).each(function() {
+          var $itemFromLeft = jQuery(this).offset().left;
+          var $itemFromRight = $viewportWidth - $itemFromLeft;
+          var $subMenuWidth = jQuery('> ul', this).outerWidth();
+          if ($subMenuWidth > $itemFromRight) {
+            $(this).addClass( 'edge' );
+          }
+          else {
+            $(this).removeClass( 'edge' );
+          }
+        });
+      }
+
       setMenu();
-      $(window).resize(setMenu);
+      menuEdgeCheck();
+
+      var onResizeActivity = Drupal.debounce(function() {
```

Comment:
Lets move this down into the `addEventListener`. I dont see a reason to add a listener to call a function to call a function to call 2 functions. It'll be easier if its all in one spot.

Reply:
done